### ISSUE TITLE
Segfault when adding more than one parameter to a custom DSP

### ISSUE THREAD

**Issue by GitGhillie** [lebedec/libfmod#18]
Hi it's me again :grimacing: 

Simplest way I've found to reproduce this: Take this example/test: https://github.com/lebedec/libfmod/blob/main/libfmod/tests/examples/core.rs

Make a new `DspParameterDesc` just like `volume_desc`.

Add it to the list of parameters: `paramdesc: vec![volume_desc, other_desc]`.

Run and it causes a segfault.

I've tried multiple ways to set it up with the underlying FMOD_DSP_DESCRIPTION and FMOD_DSP_PARAMETER_DESC but I couldn't get anything to work... If you have any tips to troubleshoot such an issue that would already be much appreciated!

**Comment by lebedec** [comment]
Oh.. It looks like incorrect handling of an array of structs. There is a terrible implementation of `vec_as_mut_ptr!` used for this =) 

Workaround should be something like this:

```Rust
let paramdesc: Box<[FMOD_DSP_PARAMETER_DESC]> = Box::new([volume_desc.into(), other_desc.into()]);
let paramdesc: *mut [FMOD_DSP_PARAMETER_DESC] = Box::into_raw(paramdesc);

let dspdesc = ffi::FMOD_DSP_DESCRIPTION {
	paramdesc: paramdesc,
	... 
}
```
After calling `Box::into_raw`, the FMOD is responsible for the memory of DPS parameters previously managed by Rust. 

This is a critical bug. I will work on it when I have time.

**Comment by lebedec** [comment]
fixed in `2.222.4`

**Comment by GitGhillie** [comment]
Thank you for the quick resolution!! That `into_boxed_slice()` is really useful...
It shouldn't be long now for me to complete my plugin :)
